omega/DecisionTree/DecisionTreeClassifier.py

```diff
@@ -0,0 +1,230 @@
+
+        Parameters
+        ----------
+        x : training data
+        y : training labels
+
+        Returns
+        -------
+        A dictionary containing the tree.
+        """
+        root = self.__get_split(x, y)
+        self.__split(root, 1)
+        return root
+
+    def __get_split(self, x, y):
+        """Finds the best split given the training data and labels
+
+        Parameters
+        ----------
+        x : training data
+        y : training labels
+
+        Returns
+        -------
+        A node for the tree with best split value given the data.
+        """
+        b_index, b_value, b_score, b_groups = 999, 999, 999, None
+
+        classes = list()
+        for val in y:
+            if val not in classes:
+                classes.append(val)
+
+        for index in range(x.size()[1]):
+            for row in x:
+                groups = self.__test_split(index, row[index], x, y)
+                score = self.__gini_index(groups[0], groups[1], classes)
+                if score < b_score:
+                    b_index, b_value, b_score, b_groups = index, row[index], score, groups
+        return {'index': b_index, 'value': b_value, 'groups': b_groups}
+
+    def __to_terminal(self, group):
+        """Determines the label of a terminal node.
+
+        Parameters
+        ----------
+        group : list
+            list contains two elements, data and labels of the rows associated with the split.
+
+        Returns
+        -------
+        target label of a terminal node.
+        """
+        res = torch.mode(group[1])[0][0]
+        return res
+
+    def __split(self, node, depth):
+        """Build the subtree given a node, keeps track of maximum depth of the tree.
+
+        Parameters
+        ----------
+        node : node for which the subtree is built
+        depth : depth of the node
+        """
+        left, right = node['groups']
+        del(node['groups'])
+        if not left[1].size() or not right[1].size():
+            group = left if left[1].size() else right
+            node['left'] = node['right'] = self.__to_terminal(group)
+            return
+
+        if depth >= self.__max_depth:
+            node['left'] = self.__to_terminal(left)
+            node['right'] = self.__to_terminal(right)
+            return
+
+        if left[1].size()[0] < self.__min_size:
+            node['left'] = self.__to_terminal(left)
+        else:
+            node['left'] = self.__get_split(left[0], left[1])
+            self.__split(node['left'], depth + 1)
+
+        if right[1].size()[0] < self.__min_size:
+            node['right'] = self.__to_terminal(right)
+        else:
+            node['right'] = self.__get_split(right[0], right[1])
+            self.__split(node['right'], depth + 1)
+
+    def predict(self, row):
+        """Given a row as data, predicts the class of the dataset.
+
+        Parameters
+        ----------
+        row : row similar to the training data
```

Comment:
Sphinx will not process this docstring, check the docstrings in my code and follow
https://pythonhosted.org/an_example_pypi_project/buildanduploadsphinx.html